DOC: fix PR09,PR08 doc string errors in Timestamp, NaT, Timedelta class

josibake opened the pull request:
This fixes PR09,PR08 errors in Timestamp, NaT, Timedelta classes

Example:

```
pandas.Timestamp.ceil: Parameter "ambiguous" description should finish with "."
pandas.Timestamp.ceil: Parameter "nonexistent" description should finish with "."
pandas.Timestamp.day_name: Parameter "locale" description should finish with "."
pandas.Timestamp.floor: Parameter "ambiguous" description should finish with "."
pandas.Timestamp.floor: Parameter "nonexistent" description should finish with "."
pandas.Timestamp.fromordinal: Parameter "ordinal" description should finish with "."
pandas.Timestamp.fromordinal: Parameter "freq" description should finish with "."
pandas.Timestamp.month_name: Parameter "locale" description should finish with "."
pandas.Timestamp.now: Parameter "tz" description should finish with "."
pandas.Timestamp.round: Parameter "ambiguous" description should finish with "."
pandas.Timestamp.round: Parameter "nonexistent" description should finish with "."
pandas.Timestamp.today: Parameter "tz" description should finish with "."
pandas.Timestamp.tz_localize: Parameter "ambiguous" description should finish with "."
pandas.Timestamp.tz_localize: Parameter "nonexistent" description should finish with "."
pandas.Timestamp.tz_localize: Parameter "errors" description should finish with "."
```

closes #28673

- [x] closes #xxxx
- [x] tests added / passed

Commit:
DOC: fix PR09 doc string errors in Timestamp class

This fixes Parameter {} description should finish with "."
in the Timestamp class

closes #28673

Commit:
DOC: fix PR09 doc string errors in Timestamp class

This fixes Parameter {} description should finish with "."
in the Timestamp class

closes #28673

josibake commented:
looks like the tests are failing due to:

```
compare = (<class 'pandas._libs.tslibs.timestamps.Timestamp'>, 'ceil')

    @pytest.mark.parametrize(
        "compare",
        (
            _get_overlap_public_nat_methods(Timestamp, True)
            + _get_overlap_public_nat_methods(Timedelta, True)
        ),
    )
    def test_nat_doc_strings(compare):
        # see gh-17327
        #
        # The docstrings for overlapping methods should match.
        klass, method = compare
        klass_doc = getattr(klass, method).__doc__
```

which i am assuming means i will also need to edit the Timedelta class

josibake commented:
adding NaT, Timedelta class since it looks like I'll have to fix those too to get it to past the tests

Commit:
fix docstrings for Timestamp, NaT classes

docstrings for Timestamp, NaT and Timedelta classes must match
for overlapping methods. verified they now pass the test_nat.py test

Commit:
fix docstrings for Timestamp, NaT classes

docstrings for Timestamp, NaT and Timedelta classes must match
for overlapping methods. verified they now pass the test_nat.py test

Commit:
fix unit description in Timedelta

Commit:
fix unit description in Timedelta

josibake commented:
confirmed that this fixes PR09 for `Timedelta,Timestamp,NaT` classes:

```
 ~/pandas-josibake/scripts/validate_docstrings.py --errors=PR09 | grep -E "Timedelta\.|Timestamp\.|NaT\."
```

josibake commented:
this is related to #27977 and #28602 (adding for visibility)

datapythonista commented on pandas/_libs/tslibs/nattype.pyx:
```diff
@@ -547,113 +547,122 @@ class NaTType(_NaT):
-default 'raise'
+        freq : str
+            Frequency string indicating the rounding resolution.
+        ambiguous : {'raise', 'NaT', bool}, default 'raise'
```
Not sure if `bool or {'raise', 'NaT'}` may make more sense.

josibake replied:
It did feel awkward using bool (as opposed to something like `{'raise', 'NaT', True, False}`). I agree, though, `bool or {'raise', 'NaT'}` makes the most sense.

datapythonista commented on pandas/_libs/tslibs/nattype.pyx:
```diff
@@ -547,113 +547,122 @@ class NaTType(_NaT):
+            * 'NaT' will return NaT for an ambiguous time.
+            * 'raise' will raise an AmbiguousTimeError for an ambiguous time.
+
+            .. versionadded:: 0.24.0.
```
I think the period at the end is to make the script happy, but shouldn't be there. Better remove it, and we'll fix the script (feel free to open an issue and work in a PR if you want)

josibake replied:
just checked the validate script: it does actually know to strip out the sphinx directives before checking for a period. turns out I was being overzealous 😅

datapythonista commented on pandas/_libs/tslibs/nattype.pyx:
```diff
@@ -547,113 +547,122 @@ class NaTType(_NaT):
+            * 'raise' will raise an AmbiguousTimeError for an ambiguous time.
+
+            .. versionadded:: 0.24.0.
+        nonexistent : {'raise', 'shift_forward', 'shift_backward, 'NaT', \
```
Did you build the html to see if this renders ok?

josibake replied:
I checked for `pandas.Timestamp` and everything looked good. Couldn't figure out how to build it for just the `NaTType` class, but it should be fine considering its the same docstring as the Timestamp class.

datapythonista commented on pandas/_libs/tslibs/nattype.pyx:
```diff
@@ -547,113 +547,122 @@ class NaTType(_NaT):
+              nonexistent times.
 
-            .. versionadded:: 0.24.0
+            .. versionadded:: 0.24.0.
```
Same about period.

WillAyd replied:
Hmm does this render correctly? I don't think we should have a period on a `versionadded`

datapythonista reviewed (commented):
Thanks for working on this, great PR. Added few minor comments, but looks good.

Commit:
change ambiguous param args and remove period

change ambiguous parameter to be more readable
remove the period after versionadd

Commit:
fix merge conflict

josibake commented:
sorry about re-adding all the commits @datapythonista ; something was screwed up with how I had my local branch set up. should be fixed now.

WillAyd commented on pandas/_libs/tslibs/nattype.pyx:
```diff
@@ -547,113 +547,122 @@ class NaTType(_NaT):
-            - timedelta objects will shift nonexistent times by the timedelta
-            - 'raise' will raise an NonExistentTimeError if there are
-              nonexistent times
+            * 'shift_forward' will shift the nonexistent time forward to the
```
Personal nit but I think it looks weird to add periods at the end of bullet points. The only one that technically needs this would be the last, at least until #25786 goes through

josibake replied:
also wasn't happy about periods on bullet points, but liked it better with all having periods vs last one only. considering the fix to the validation script is about to go through, wouldn't it be better to just take out periods for all bullet points and let them keep failing until the validation script is finished?

WillAyd replied:
Would prefer just the last to have one for now but it's not a blocker for me if they all do. Allowing to fail won't work though

josibake replied:
ill leave them all for right now, just its more consistent with other parts of the docs as of right now

WillAyd commented on pandas/_libs/tslibs/nattype.pyx:
```diff
@@ -547,113 +547,122 @@ class NaTType(_NaT):
+              nonexistent times.
 
-            .. versionadded:: 0.24.0
+            .. versionadded:: 0.24.0.
```
Same here

WillAyd commented on pandas/_libs/tslibs/timestamps.pyx:
```diff
@@ -779,35 +788,42 @@ default 'raise'
               into the specified timezone. Use ``nonexistent='NaT'`` instead.
 
-              .. deprecated:: 0.24.0
+            .. deprecated:: 0.24.0
```
Was there a reason for unindenting this (applicable in a few places)? I think this changes the meaning of just the `coerce` literal being deprecated to the entire `errors` argument being deprecated

josibake replied:
good catch, that was a mistake. not sure why i moved those, but it definitely changes the meaning.

jbrockmendel committed:
CLN: Assorted typings (#28604)

jb8429 committed:
DOC: Fixed PR08 docstring errors in pandas.tseries (#28571)

Commit:
remove periods after versionadd and fix deprecated directive

missed a few periods from the last commit
also incorrectly moved a sphinx directive earlier

TomAugspurger committed:
DOC: Add scaling to large datasets section (#28577)

* DOC: Add scaling to large datasets section

Closes https://github.com/pandas-dev/pandas/issues/28315

charlesdong1991 committed:
BUG: restore limit in RangeIndex.get_indexer (#28671)

ladyyvii committed:
DOC: Fixed PR09 docstring errors in pandas.tseries (#27977) (#28707)

ladyyvii committed:
CI Failing: TestReadHtml.test_spam_url #28708 (#28710)

jbrockmendel committed:
TST: un-xfail incorrectly xfailed tests for maybe_promote (#28564)

jbrockmendel committed:
CLN: Define and pin GroupBy properties without exec (#28651)

dayoreke committed:
DOC: Fixed PR08, PR09 doctring issues in pandas.core.groupby (#28709)

LauraCollard committed:
DOC: Fixed PR06 docstring errors in pandas.DataFrame (#28718)

WillAyd commented on pandas/core/algorithms.py:
```diff
@@ -1935,13 +1933,15 @@ def diff(arr, n, axis=0):
-        lag_indexer = tuple(lag_indexer)
+        # To keep mypy happy, _res_indexer is a list while res_indexer is
+        #  a tuple, ditto for lag_indexer.
+        _res_indexer = [slice(None)] * arr.ndim
```
I don't think this is supposed to be part of your PR. Are you sure you merged in the latest changes from upstream/master?

LauraCollard committed:
DOC: Fixed PR06 docstrings errors in pandas.arrays.IntervalArray (#28721)

Commit:
DOC: fix PR09,PR08 docstring errors in pandas.plotting (#28689)

tolaa001 committed:
precursor to Split out test_pytables.py to sub-module of tests (#28715)

Commit:
fix merge conflict

Commit:
Merge remote-tracking branch 'upstream/master' into PR09-Timestamp-fix

josibake commented:
apologies, i am a little confused on this one and cant figure out what im doing wrong. I ran:

```
git checkout PR09-Timestamp-fix
git fetch upstream
git merge upstream/master
```

but this adds more commits into the PR. let me read up on github docs and figure out what im doing and then ill take a another stab at opening a PR
